cmake/projects/Boost/schemes/url_sha1_boost.cmake.in

```diff
@@ -36,45 +36,60 @@ if("@MSVC@")
 endif()
 
 # Same logic in all url_sha1_boost* schemes (TODO: DRY)
-macro(BOOST_COMPONENT name version)
+macro(BOOST_COMPONENT_SINCE name version)
   list(APPEND BOOST_COMPONENT_NAMES ${name})
   set(BOOST_COMPONENT_${name}_VERSION ${version})
 endmacro()
 
-boost_component(atomic 1.53.0)
-boost_component(chrono 1.47.0)
-boost_component(container 1.56.0)
-boost_component(context 1.51.0)
-boost_component(coroutine 1.53.0)
-boost_component(coroutine2 1.60.0)
-boost_component(date_time 1.29.0)
-boost_component(exception 1.36.0)
-boost_component(fiber 1.62.0)
-boost_component(filesystem 1.30.0)
-boost_component(graph 1.18.0)
-boost_component(graph_parallel 1.18.0)
-boost_component(iostreams 1.33.0)
-boost_component(locale 1.48.0)
-boost_component(log 1.54.0)
-boost_component(math 1.23.0)
-boost_component(metaparse 1.61.0)
-boost_component(mpi 1.35.0)
-boost_component(program_options 1.32.0)
-boost_component(python 1.19.0)
-boost_component(random 1.15.0)
-boost_component(regex 1.18.0)
-boost_component(serialization 1.32.0)
-boost_component(signals 1.29.0)
-boost_component(system 1.35.0)
-boost_component(test 1.21.0)
-boost_component(thread 1.25.0)
-boost_component(timer 1.9.0)
-boost_component(type_erasure 1.60.0)
-boost_component(wave 1.33.0)
+# set boost component valid version range
+# - version_since: first boost version the component is include
+# - version_until: first boost version the component isn't included anymore
+macro(BOOST_COMPONENT_RANGE name version_since version_until)
+  list(APPEND BOOST_COMPONENT_NAMES ${name})
+  set(BOOST_COMPONENT_${name}_VERSION ${version_since})
+  set(BOOST_COMPONENT_${name}_VERSION_UNTIL ${version_until})
+endmacro()
+
+boost_component_since(atomic 1.53.0)
```

Comment:
Let's create function with signature `boost_component(COMPONENT <component> SINCE <v1> UNTIL <v2>)`. `COMPONENT` and `SINCE` required, `UNTIL` is optional.